aiohttp/web_urldispatcher.py

```diff
@@ -1050,6 +1084,32 @@ def register_resource(self, resource: AbstractResource) -> None:
             self._named_resources[name] = resource
         self._resources.append(resource)
 
+        if isinstance(resource, MatchedSubAppResource):
+            # We cannot index match sub-app resources because they have match rules
+            self._matched_sub_app_resources.append(resource)
+        else:
+            self.index_resource(resource)
+
+    def _get_resource_index_key(self, resource: AbstractResource) -> str:
+        """Return a key to index the resource in the resource index."""
+        canonical = resource.canonical
+        if "{" in canonical:  # strip at the first { to allow for variables
+            canonical = canonical.partition("{")[0].rstrip("/")
```

Comment:
This works without the if?
```suggestion
        # strip at the first { to allow for variables
        canonical = resource.canonical.partition("{")[0].rstrip("/")
```